### PR TITLE
Add Skeleton Lake testnet

### PR DESCRIPTION
**Pull request by ralexstokes**
Part of the eth2 interop cleanup.

Adds the SkeletonLake testnet used for early interop experiments.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by ralexstokes** [comment]
Missing some commits:

https://github.com/ethereum/trinity/commit/26417da4a819213ab1b96044a1c2f2f82507a26f
https://github.com/ethereum/trinity/commit/cb5ea140a3668e7f0887bb65443c7d2408b51aec
https://github.com/ethereum/trinity/commit/6c0b8c8e1584e51b704d04151988727550265f74

**Comment by ralexstokes** [comment]
Should merge in:

[p2p cancel] https://github.com/ethereum/trinity/pull/1168
[ms-reso] https://github.com/ethereum/trinity/pull/1169
[crosslinks] https://github.com/ethereum/trinity/pull/1170
[type printing] https://github.com/ethereum/trinity/pull/1171
[add dockerfile] https://github.com/ethereum/trinity/pull/1172
[eth2 genesis utils] https://github.com/ethereum/trinity/pull/1173

and rebase before merging in this PR

File: trinity/plugins/registry.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import pkg_resources
```
Comment (pipermerriam) [inline trinity/plugins/registry.py]: This file was renamed to be under `trinity/components/registry`

Comment (ralexstokes) [reply trinity/plugins/registry.py]: yep -- thanks for pointing it out -- its on my list to fix this before merging this in

**Comment by ralexstokes** [comment]
before we merge this in, figure out if we can remove the introduce of config and keystore files in #1173.

File: README-INTEROP [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   $ brew install leveldb snappy  # If you're using osx
+   $ pip install -e .[eth2-dev]
+
+5. You should now be ready to run trinity!
+   For the purposes of interop trinity hard-codes some files, so it must be run with a cwd
+   of the repository root.
+
+   Here's an exapmle of how to run it:
+
+   $ PYTHONWARNINGS=ignore::DeprecationWarning trinity-beacon -l DEBUG interop --validators 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15 --start-delay 20 --wipedb
```
[the remark this answers is not in the document]
Comment (ralexstokes) [reply README-INTEROP]: i'll investigate, thanks for reporting :)

File: eth2/beacon/state_machines/forks/serenity/block_processing.py [hunk 3]
```diff
@@ -101,6 +106,7 @@ def process_block(
     config: Eth2Config,
     check_proposer_signature: bool = True,
 ) -> BeaconState:
+    # logger.debug('process_block called')
```
[the remark this answers is not in the document]
Comment (ralexstokes) [reply eth2/beacon/state_machines/forks/serenity/block_processing.py]: i think we will just drop these...

File: eth2/beacon/state_machines/forks/serenity/epoch_processing.py [hunk 4]
```diff
@@ -476,12 +480,18 @@ def process_rewards_and_penalties(
...
     rewards_for_attestations, penalties_for_attestations = get_attestation_deltas(
         state, config
     )
+    # logger.debug(f'rewards1: {rewards_for_attestations}')
+    # logger.debug(f'penalties1: {penalties_for_attestations}')
     rewards_for_crosslinks, penalties_for_crosslinks = get_crosslink_deltas(
         state, config
     )
+    # logger.debug(f'rewards2: {rewards_for_crosslinks}')
+    # logger.debug(f'penalties2: {penalties_for_crosslinks}')
```
[the remark this answers is not in the document]
Comment (ralexstokes) [reply eth2/beacon/state_machines/forks/serenity/epoch_processing.py]: ditto ditto

File: README-INTEROP (hunk 2 above)
[the remark this answers is not in the document]
Comment (ralexstokes) [reply README-INTEROP]: the global `override_lengths` was at it again... fixed now :)

**Comment by ralexstokes** [comment]
CI failures are known issues w/ `trinity-eth1`, going to merge this in to move ahead w/ SkeletonLake testnet

File: Dockerfile.eth2 [hunk 5]
```diff
@@ -10,9 +10,10 @@ RUN apt-get -y install libsnappy-dev gcc g++ cmake
...
 RUN git clone https://github.com/ethereum/trinity.git .
-RUN git checkout $GITREF && pip install -e .[dev] --no-cache-dir
+RUN git checkout $GITREF
+RUN pip install -e .[dev] --no-cache-dir
 RUN pip install -U trinity --no-cache-dir
 
 EXPOSE 30303 30303/udp
 # Trinity shutdowns aren't yet solid enough to avoid the fix-unclean-shutdown
-ENTRYPOINT trinity $EXTRA_OPTS fix-unclean-shutdown && trinity-beacon $EXTRA_OPTS
+ENTRYPOINT trinity $EXTRA_OPTS fix-unclean-shutdown && trinity-beacon $EXTRA_OPTS
```
[the remark this answers is not in the document]
Comment (ralexstokes) [reply Dockerfile.eth2]: you mean at the end of the file, right?

[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply Dockerfile.eth2]: Exactly

**Pull request merged**